src/repl/scala/tools/nsc/interpreter/MemberHandlers.scala

```diff
@@ -214,27 +214,34 @@ trait MemberHandlers {
     val Import(expr, selectors) = imp
     def targetType = intp.global.rootMirror.getModuleIfDefined("" + expr) match {
       case NoSymbol => intp.typeOfExpression("" + expr)
-      case sym      => sym.thisType
+      case sym      => sym.moduleClass.thisType
     }
-    private def importableTargetMembers = importableMembers(targetType).toList
+    private def importableTargetMembers = importableMembers(exitingTyper(targetType)).toList
     // wildcard imports, e.g. import foo._
     private def selectorWild    = selectors filter (_.name == nme.USCOREkw)
-    // renamed imports, e.g. import foo.{ bar => baz }
-    private def selectorRenames = selectors map (_.rename) filterNot (_ == null)
+
+    // non-wildcard imports
+    private def individualSelectors = selectors filter { x =>
+      x.name != nme.USCOREkw && x.name != null && x.rename != nme.USCOREkw & x.rename != null
```

Comment:
Could we encapsulate this stuff in `Contexts`, say? Looks like we already have `warnUnusedImports`, which compares against the more logical `nme.WILDCARD`, by the way.